src/string/lipogram.rs

```diff
@@ -0,0 +1,80 @@
+use std::collections::HashSet;
+
+///*
+/// Function that returns the letters that are missing from the input slice  
+/// and are present in the English alphabet  
+///
+/// ## Arguments
+///
+/// * `in_str` - the slice that will be checked for missing characters
+///
+fn compute_missing(in_str: &str) -> HashSet<char> {
+    let alphabet: HashSet<char> = "abcdefghijklmnopqrstuvwxyz".chars().collect();
+
+    let letters_used: HashSet<char> = in_str
+        .to_lowercase()
+        .chars()
+        .filter(|c| c.is_ascii_alphabetic())
+        .collect();
+
+    alphabet.difference(&letters_used).cloned().collect()
+}
+
+//*
+/// Function that checks if the slice is a lipogram with specific missing letters.  
+/// Lipogram - sentence in which a particular letter or group of letters is avoided
+///
+/// ## Arguments
+///
+/// * `lipogram_str` - the slice that will be checked if is a lipogram with specific missing letters
+/// * `missing_chars` - the characters that has to be missing
+///
+/// ## Examples
+///
+/// ```
+/// use the_algorithms_rust::string::is_lipogram;
+/// use std::collections::HashSet;
+///
+/// assert!(
+///    !is_lipogram("The quick brown fox jumps over the lazy dog",
+///    &HashSet::from(['x'])
+/// ));
+///
+/// assert!(
+///    is_lipogram("The brown cat jumped over the lazy dog with a brick",
+///    &HashSet::from(['f', 'q', 's', 'x'])
+/// ));
+///
+/// assert!(
+///    !is_lipogram("The quick brown fox jumped over the lazy dog",
+///    &HashSet::from(['x'])
+/// ));
+/// ```
+pub fn is_lipogram(lipogram_str: &str, missing_chars: &HashSet<char>) -> bool {
+    missing_chars == &compute_missing(&lipogram_str.to_lowercase())
```

Comment:
This is strange, cf. https://github.com/TheAlgorithms/Rust/blob/ea65e7ffab3508eae87249360701a70534696336/src/string/lipogram.rs#L15.

I think what you meant is something like:
```suggestion
    if !missing_chars.iter().all(|&c| c.is_lowercase()) {
        panic!("missing_chars should be all lowercase.")
    }

    missing_chars == &compute_missing(lipogram_str)
```

Please have a look at the suggestion regarding the tests: it contains some new ones covering _lower and uppercase letters_.